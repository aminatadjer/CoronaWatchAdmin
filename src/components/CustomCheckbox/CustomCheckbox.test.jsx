import React from 'react';
import ReactDom from 'react-dom';
import CustomCheckbox from './CustomCheckbox';
import { render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', ()=>{
    const div = document.createElement('div');
    ReactDom.render(<CustomCheckbox></CustomCheckbox>, div);
});

it('renders CustomButton correctly', ()=>{
    const {getByTestId} = render(<CustomCheckbox isChecked={true}></CustomCheckbox>)
    expect(getByTestId('customchck')).toBeChecked;
});

it('renders CustomButton correctly', ()=>{
    const {getByTestId} = render(<CustomCheckbox label="This is a checkbox"></CustomCheckbox>)
    expect(getByTestId('customchck')).toHaveTextContent("This is a checkbox");
});

it('renders CustomButton correctly', ()=>{
    const {getByTestId} = render(<CustomCheckbox inline={true}></CustomCheckbox>)
    expect(getByTestId('customchck')).toHaveClass('checkbox-inline');
});

// Snapshots 

it ('matches snapshot', ()=>{
    const tree = renderer.create(<CustomCheckbox isChecked={true}></CustomCheckbox>).toJSON();
    expect(tree).toMatchSnapshot();
})

it ('matches snapshot', ()=>{
    const tree = renderer.create(<CustomCheckbox label="This is a checkbox"></CustomCheckbox>).toJSON();
    expect(tree).toMatchSnapshot();
})

it ('matches snapshot', ()=>{
    const tree = renderer.create(<CustomCheckbox number={1}></CustomCheckbox>).toJSON();
    expect(tree).toMatchSnapshot();
})

it ('matches snapshot', ()=>{
    const tree = renderer.create(<CustomCheckbox inline={true}></CustomCheckbox>).toJSON();
    expect(tree).toMatchSnapshot();
})