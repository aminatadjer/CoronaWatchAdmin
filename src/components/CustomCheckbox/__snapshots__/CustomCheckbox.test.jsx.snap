// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="checkbox"
  data-testid="customchck"
>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  <label />
</div>
`;

exports[`matches snapshot 2`] = `
<div
  className="checkbox"
  data-testid="customchck"
>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <label>
    This is a checkbox
  </label>
</div>
`;

exports[`matches snapshot 3`] = `
<div
  className="checkbox"
  data-testid="customchck"
>
  <input
    checked={false}
    id={1}
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor={1}
  />
</div>
`;

exports[`matches snapshot 4`] = `
<div
  className="checkbox checkbox-inline"
  data-testid="customchck"
>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <label />
</div>
`;
